sdk/src/compressed_account.rs

```diff
@@ -1,421 +1,215 @@
-    ) -> Result<Self>;
-    fn new_address_params(&self) -> Vec<NewAddressParamsPacked>;
-    fn input_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<PackedCompressedAccountWithMerkleContext>>;
-    fn output_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<OutputCompressedAccountWithPackedContext>>;
+pub trait LightAccounts<'a>: Sized {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self>;
 }
 
+// TODO(vadorovsky): Implment `LightAccountLoader`.
+
 /// A wrapper which abstracts away the UTXO model.
-pub enum LightAccount<T>
+pub struct LightAccount<'info, T>
 where
-    T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
+    T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Default + Discriminator,
 {
-    Init(LightInitAccount<T>),
-    Mut(LightMutAccount<T>),
-    Close(LightCloseAccount<T>),
+    /// State of the output account which can be modified by the developer in
+    /// the program code.
+    account_state: T,
+    /// Account information.
+    account_info: LightAccountInfo<'info>,
+    /// Hash of the input account.
+    input_hash: Option<[u8; 32]>,
 }
 
-impl<T> LightAccount<T>
+impl<'info, T> LightAccount<'info, T>
 where
     T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Default + Discriminator,
 {
-    pub fn new_init(
-        merkle_context: &PackedMerkleContext,
-        address_merkle_context: &PackedAddressMerkleContext,
-        address_merkle_tree_root_index: u16,
-    ) -> Self {
-        Self::Init(LightInitAccount::new(
-            merkle_context,
-            address_merkle_context,
-            address_merkle_tree_root_index,
-        ))
+    pub fn from_meta_init(
+        meta: &'info LightAccountMeta,
+        discriminator: [u8; 8],
+        new_address: [u8; 32],
+        new_address_seed: [u8; 32],
+        program_id: &Pubkey,
+    ) -> Result<Self> {
+        let account_state = T::default();
+        let account_info = LightAccountInfo::from_meta_init(
+            meta,
+            discriminator,
+            new_address,
+            new_address_seed,
+            program_id,
+        )?;
+        Ok(Self {
+            account_state,
+            account_info,
+            input_hash: None,
+        })
     }
 
-    pub fn try_from_slice_mut(
-        v: &[u8],
-        merkle_context: &PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: &PackedAddressMerkleContext,
+    pub fn from_meta_mut(
+        meta: &'info LightAccountMeta,
+        discriminator: [u8; 8],
+        program_id: &Pubkey,
     ) -> Result<Self> {
-        Ok(Self::Mut(LightMutAccount::try_from_slice(
-            v,
-            merkle_context,
-            merkle_tree_root_index,
-            address_merkle_context,
-        )?))
+        let account_info = LightAccountInfo::from_meta_mut(meta, discriminator, program_id)?;
```

Comment:
this clones the input data right to `LightAccountInfo.data` which we don't use in this context.
Should we add `LightAccountInfo::from_meta` which sets `LightAccountInfo.data` to `None`?